### PR TITLE
Refactor RFC3339 helpers, tighten date codec parsing, and add Date.offset tests

### PR DESCRIPTION
**Pull request by davidpcaldwell**
## Summary
- Refactor shared RFC3339 helper logic used by time codecs
- Tighten Date RFC3339 codec parsing behavior and related tests
- Add focused Fifty coverage for `Date.offset`, including boundary cases

## Changes
- Updated [js/time/module.js](js/time/module.js)
- Updated [js/time/module.fifty.ts](js/time/module.fifty.ts)

## Test Coverage
- Ran `./fifty test.jsh js/time/module.fifty.ts`
- Suite passed

## Notes
- Branch contains the RFC3339 refactor/parsing commits plus the Date.offset test coverage commit

### CODE REVIEW

File: js/time/module.fifty.ts [hunk 1]
```diff
@@ -195,7 +155,9 @@ namespace slime.time {
 		function(
 			fifty: slime.fifty.test.Kit
 		) {
-			fifty.tests.Date = fifty.test.Parent();
+			fifty.tests.exports.Date = fifty.test.Parent();
+
+			fifty.tests.exportsDate = fifty.test.Parent();
```
[the remark this answers is not in the document]
Comment (davidpcaldwell) [reply js/time/module.fifty.ts]: Fixed.

File: js/time/module.js [hunk 2]
```diff
@@ -1010,12 +1010,45 @@
...
+		}
+
+		var isValidGregorianDate = function(year, month, day) {
+			if (!isInteger(year) || !isInteger(month) || !isInteger(day)) return false;
+			if (month < 1 || month > 12) return false;
+			if (day < 1 || day > 31) return false;
+
+			var js = new Date(Date.UTC(year, month-1, day));
+			return js.getUTCFullYear() == year && js.getUTCMonth() == month-1 && js.getUTCDate() == day;
+		}
```
[the remark this answers is not in the document]
Comment (davidpcaldwell) [reply js/time/module.js:1038]: Thanks for the catch. Addressed in 8b8d9483a.

isValidGregorianDate now avoids Date.UTC for validation and uses a UTC date initialized with setUTCFullYear(year, month-1, day), which correctly handles years 0000-0099 (no 1900 offset coercion).

I also kept the integer/month/day guard checks and re-ran ./fifty test.jsh js/time/module.fifty.ts (passing).
